src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,105 @@
+package duke;
+
+import java.nio.file.Path;
+
+import duke.command.Command;
+import duke.exception.DukeInputException;
+import duke.exception.DukeSaveDataException;
+import duke.io.InputHandler;
+import duke.io.OutputHandler;
+
+
+
+/** Contains main software logic loop and initialization logic.*/
+public class Duke {
+
+    /** TaskManager object that contains all the tasks in Duke's memory */
+    private TaskManager taskManager;
+
+    /** Ui object that handles all display and input from user */
+    private Ui ui;
+
+    /** Path object that contains the relative directory path to the save file */
+    private Path filePath;
+
+    /** SaveManager object that handles all saving and loading of save data on load or on exit */
+    private SaveManager saveManager;
+
+    /**
+     * Constructs new Duke object with default save file directory.
+     * Defaults save file directory to "/data/data.txt".
+     */
+    public Duke() {
+        this(Path.of("/data", "data.txt"));
+    }
+
+    /**
+     * Constructs new Duke object with custom save file directory.
+     *
+     * @param filePath Path object representing save file directory.
+     */
+    public Duke(Path filePath) {
+        // Initialise properties
+        this.filePath = filePath;
```

Comment:
Is the **this.** really necessary? Can be removed for better code readability

Reply:
Thanks for the suggestion, however I feel that `filePath = filePath;` appears to be a rather confusing line so it is better to be explicit with the use of `this`. Let me know if it is a code style violation though, not sure about that.